x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/components/tables/shared_columns.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { EuiTableActionsColumnType, EuiTableFieldDataColumnType } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { FormattedNumber } from '@kbn/i18n-react';
+
+import { DELETE_BUTTON_LABEL, MANAGE_BUTTON_LABEL } from '../../../../../shared/constants';
+import { EngineListDetails } from '../../types';
+
+export const ENGINE_NAME: EuiTableFieldDataColumnType<EngineListDetails> = {
```

Comment:
if `engines_table` is the only component using these columns I would not abstract them into a "shared" file yet. To me its a premature optimization. I would just leave these defined in the `engines_table.tsx`